ui_framework/components/modal/confirm_modal.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+
+import { KuiModal } from '../modal';
+import { KuiModalFooter } from './modal_footer';
+import { KuiModalHeader } from './modal_header';
+import { KuiModalBody } from './modal_body';
+import { KuiModalBodyText } from './modal_body_text';
+import { KuiButton } from '../index';
+
+export function KuiConfirmModal({ message, title, onCancel, onConfirm, cancelButtonText, confirmButtonText, ...rest }) {
+  return (
+    <KuiModal style={{ 'width': '450px' }} { ...rest }>
+      {
+        title ?
+          <KuiModalHeader>
+            <div className="kuiModalHeader__title" data-test-subj="confirmModalTitleText">
+              { title }
+            </div>
+          </KuiModalHeader>
+          : null
+      }
+      <KuiModalBody>
+        <KuiModalBodyText data-test-subj="confirmModalBodyText">
+          { message }
+        </KuiModalBodyText>
+      </KuiModalBody>
+
+      <KuiModalFooter>
+        <KuiButton
+          type="hollow"
+          data-test-subj="confirmModalCancelButton"
+          onClick={ onCancel }
+        >
+          {cancelButtonText}
+        </KuiButton>
+        <KuiButton
+          type="primary"
+          data-test-subj="confirmModalConfirmButton"
+          onClick={ onConfirm }
+        >
+          {confirmButtonText}
+        </KuiButton>
+      </KuiModalFooter>
+    </KuiModal>
+  );
+}
+
+KuiConfirmModal.propTypes = {
+  message: React.PropTypes.string,
```

Comment:
> Note: React.PropTypes is deprecated as of React v15.5. Please use the prop-types library instead.

React.propTypes has been depricated, so we shouldn't be using them in new code. Use the `prop-types` module instead.